### ISSUE TITLE
Handling scoped paths without leading slashes

### ISSUE THREAD

**Issue by ilammy** [actix/actix-web#460]
I have encountered a weird regression when upgrading actix-web from 0.6.15 to 0.7.3. Our code was using scopes to install routes with paths that _do not_ have leading slashes. This was fine in 0.6, but with 0.7 the server started replying with HTTP 404.

Here's an example which works with `actix-web = "=0.6.15"` but panics with `actix-web = "=0.7.3"`:

```rust
extern crate actix_web;

use actix_web::{http, test::TestServer, App, HttpRequest};

fn reply(_: HttpRequest) -> String {
    format!("reply")
}

fn main() {
    let mut server = TestServer::with_factory(|| {
        App::new().scope("test", |scope| {
            scope
                .route("request", http::Method::GET, reply)
                .route("request", http::Method::POST, reply)
        })
    });

    let request = server.client(http::Method::GET, "/test/request")
        .finish()
        .expect("request");
    let response = server.execute(request.send())
        .expect("response");
    assert_eq!(response.status().as_u16(), 200);

    let request = server.client(http::Method::POST, "/test/request")
        .finish()
        .expect("request");
    let response = server.execute(request.send())
        .expect("response");
    assert_eq!(response.status().as_u16(), 200);
}
```

If leading slashes are added to path components (`/test`, `/request`) then the test is successful with 0.7.3.

Another curious point here is that it fails only when there are two or more routes for the same path with different methods. If only GET or POST route is installed and tested then it is resolved just fine without leading slashes even on actix-web 0.7.3.

Are leading slashes mandatory in paths? Maybe actix-web should enforce it or make sure that they are added automatically when needed.

**Comment by fafhrd91** [comment]
This is regression. I’ll fix it in next version

**Comment by antmoth** [comment]
I would like to try fixing this.

**Comment by fafhrd91** [comment]
@estelendur you are welcome!

**Comment by antmoth** [comment]
At this point I have to either give up or ask for help - while it seems pretty obvious that the issue lies in route registration, I have not been able to understand the code well enough to figure out which piece of code needs to be changed or how. If I had a little help I would be happy to give it another go, but I don't want to hold it back from actually being fixed.

**Comment by fafhrd91** [comment]
fixed by 968c81e
